pandas/_libs/groupby.pyx

```diff
@@ -893,7 +893,7 @@ def group_last(rank_t[:, :] out,
             for j in range(K):
                 val = values[i, j]
 
-                if not checknull(val):
+                if not checknull(val) or val is None:
```

Comment:
can you add a comment here of why we are doing this

Reply:
Done